SCons/Tool/VCPkgTests.py

```diff
@@ -0,0 +1,528 @@
+        self._useCount += 1
+        if self._useCount == 1:
+            # There shouldn't be anything configured yet
+            self.assert_empty()
+
+            # Save original functions to restore later
+            self._orig_bootstrap_vcpkg = SCons.Tool.vcpkg._bootstrap_vcpkg
+            self._orig_call_vcpkg = SCons.Tool.vcpkg._call_vcpkg
+            self._orig_install_packages = SCons.Tool.vcpkg._install_packages
+            self._orig_upgrade_packages = SCons.Tool.vcpkg._upgrade_packages
+            self._origis_mismatched_version_installed = SCons.Tool.vcpkg.is_mismatched_version_installed
+            self._orig_get_package_version = SCons.Tool.vcpkg._get_package_version
+            self._orig_get_package_deps = SCons.Tool.vcpkg._get_package_deps
+            self._orig_read_vcpkg_file_list = SCons.Tool.vcpkg._read_vcpkg_file_list
+
+            # Replace the low-level vcpkg functions with our mocks
+            SCons.Tool.vcpkg._bootstrap_vcpkg = MockVCPkg._bootstrap_vcpkg
+            SCons.Tool.vcpkg._call_vcpkg = MockVCPkg._call_vcpkg
+            SCons.Tool.vcpkg._install_packages = MockVCPkg._install_packages
+            SCons.Tool.vcpkg._upgrade_packages = MockVCPkg._upgrade_packages
+            SCons.Tool.vcpkg.is_mismatched_version_installed = MockVCPkg.is_mismatched_version_installed
+            SCons.Tool.vcpkg._get_package_version = MockVCPkg._get_package_version
+            SCons.Tool.vcpkg._get_package_deps = MockVCPkg._get_package_deps
+            SCons.Tool.vcpkg._read_vcpkg_file_list = MockVCPkg._read_vcpkg_file_list
+
+    def release(self):
+        """Called to release a ref-count on the singleton MockVCPkg object. When this hits zero, the MockVCPkg instance will tear itself down"""
+        assert(self._useCount > 0)
+        self._useCount -= 1
+        if self._useCount == 0:
+            # There shouldn't be any configuration still remaining
+            self.assert_empty()
+
+            # Restore original functions
+            SCons.Tool.vcpkg._bootstrap_vcpkg = self._orig_bootstrap_vcpkg
+            SCons.Tool.vcpkg._call_vcpkg = self._orig_call_vcpkg
+            SCons.Tool.vcpkg._install_packages = self._orig_install_packages
+            SCons.Tool.vcpkg._upgrade_packages = self._orig_upgrade_packages
+            SCons.Tool.vcpkg.is_mismatched_version_installed = self._origis_mismatched_version_installed
+            SCons.Tool.vcpkg._get_package_version = self._orig_get_package_version
+            SCons.Tool.vcpkg._get_package_deps = self._orig_get_package_deps
+            SCons.Tool.vcpkg._read_vcpkg_file_list = self._orig_read_vcpkg_file_list
+
+            # Finally, free the singleton
+            MockVCPkg.__instance = None
+
+
+    #
+    # State modification functions used by contextmanager functions below
+    #
+
+    def addAvailablePackage(self, name, version, dependencies):
```

Comment:
> __[Flake8]__ expected 1 blank line, found 0 ([view](https://sider.review/gh/repos/104670160/pulls/4226#issue-8146504))
>
> <details>
>
> | Rule |
> | ---- |
> | `E301` |
>
> You can [close this issue](https://sider.review/gh/repos/104670160/pulls/4226#issue-8146504) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

I don't understand this flake8 complaint. Is it asking me to put a blank line after "def addAvailablePackage(...):"?

Reply:
If you google "flake8 e301" (The error indicated), you'll usually find a decent explanation.
see: https://www.flake8rules.com/rules/E301.html (for this case)

Reply:
I did find that page...but it was unhelpful in this case. As you can see, there is a blank line before "def addAvailablePackage" and a blank line after the end of the function. So I don't know what to do to appease this rule...

Reply:
We can mark it to skip, I don't see anything there to complain about

Reply:
I closed a couple of the sider issues. The remaining ones are suggesting using def instead of lambda. @mwichmann - thoughts?

Reply:
I don't personally have a strong opinion.  The thinking these days seems to be if you want to use an anonymous function do it; if you're going to assign a name to it then use a real function instead - that's pretty much what the error message says anyway. lambdas work well for things like sort keys, so they definitely still have a place, it's just a smaller place than it used to be.

Reply:
My inclination is that using "def" here is the wrong thing. I'm not just using the lambda to avoid declaring an extra function. I'm also using the fact that lambdas can capture state from the surrounding context (in this case, the suffix_filters variable). I would vote to leave this as-is.

If you feel strongly, the other options I see are:
1. Convert this into a single def, containing the entire "if...else" block on lines 527-532. Slightly less performant in the case where the user wants to enumerate all files without a filter, but that's probably not the common case.
2. Convert this into 3 defs (one for each case), assign the appropriate one to a local var, and then pass the suffix_filters variable as an argument when invoking. Feels clunky to me.

Let me know what you prefer.